### PR TITLE
vdk-dag: DAGs propagate their execution type to their component jobs

### PR DESCRIPTION
**Pull request by yonitoo**
What:
Currently, jobs ran by DAGs are labelled manual executions. This is due to the execution being triggered through the Execution API. Ideally, we would like that if a DAGs is a manual execution, then all component jobs are also manual execution, and likewise for scheduled runs. This change achieves this.

Why: To achieve consistency with the data jobs' execution type.

Testing Done: all tests are passing, introduced unit test

Signed-off-by: Yoan Salambashev <ysalambashev@vmware.com>

### CODE REVIEW

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/cached_data_job_executor.py [hunk 1]
```diff
@@ -138,6 +141,19 @@ def status(self, job_name: str) -> str:
...
+        Gets the execution type of a job.
+
+        :param execution_id: the execution id of the job
+        :param team_name: the name of the owning team
+        :param job_name: the name of the job
+        :return: the job execution type (manual/scheduled)
+        """
+        details = self._executor.details_job(job_name, team_name, execution_id)
+        log.debug(f"Job execution type: {details.get('type')}")
+        return details.get("type", "manual")
```
Comment (gageorgiev) [inline projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/cached_data_job_executor.py:155]: leave a comment mentioning that we default to "manual" for clarity

**Review by gageorgiev — APPROVED** [review]

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py [hunk 2]
```diff
@@ -47,6 +53,25 @@ def __init__(self, team_name: str, dags_config: DagPluginConfiguration):
...
+                    + job_name
+                )
+            except ApiException as e:
+                if e.status == 404:
+                    log.debug(
+                        f"Local job runs return 404 status when getting the execution type: {e}"
+                    )
+                else:
+                    log.info(
+                        f"Unexpected error while checking for job execution type: {e}"
```
Comment (tozka) [inline projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py:70]: same comment as above. Review other logs and make sure there is context that may be helpful when root causing an issue.

File: projects/vdk-plugins/vdk-dag/tests/test_dag.py [hunk 3]
```diff
@@ -350,7 +406,25 @@ def test_dag_concurrent_running_jobs_limit(self):
             # assert that all the jobs finished successfully
             assert len(running_jobs) == 0
             self.httpserver.stop()
-    """
+
+    def test_dag_execution_type_propagation(self):
```
Comment (tozka) [inline projects/vdk-plugins/vdk-dag/tests/test_dag.py:410]: This is testing the happy path. Are there tests covering the error paths?

**Review by tozka — APPROVED** [review]
LGTM. I made some comments that would be nice to address. But it's approved so ship it when you feel it's ready.

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py (hunk 2 above)
Comment (yonitoo) [reply projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py]: Done.

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py [hunk 4]
```diff
@@ -47,6 +53,25 @@ def __init__(self, team_name: str, dags_config: DagPluginConfiguration):
...
+            try:
+                self._started_by = (
+                    self._job_executor.execution_type(job_name, team_name, execution_id)
+                    + "/"
+                    + job_name
+                )
+            except ApiException as e:
+                if e.status == 404:
+                    log.debug(
+                        f"Local job runs return 404 status when getting the execution type: {e}"
```
[the remark this answers is not in the document]
Comment (yonitoo) [reply projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py]: Done.

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/cached_data_job_executor.py (hunk 1 above)
Comment (yonitoo) [reply projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/cached_data_job_executor.py:156]: Done.

File: projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py [hunk 5]
```diff
@@ -47,6 +53,25 @@ def __init__(self, team_name: str, dags_config: DagPluginConfiguration):
...
+                    )
+                else:
+                    log.info(
+                        f"Unexpected error while checking for job execution type: {e}"
+                    )
+                self._started_by = f"manual/{job_name}"
+        else:
+            self._started_by = "manual/default"
```
[the remark this answers is not in the document]
Comment (yonitoo) [reply projects/vdk-plugins/vdk-dag/src/vdk/plugin/dag/dag.py:81]: You are right, changed it.

File: projects/vdk-plugins/vdk-dag/tests/test_dag.py (hunk 3 above)
Comment (yonitoo) [reply projects/vdk-plugins/vdk-dag/tests/test_dag.py:410]: We decided to leave this one for later.

**Pull request merged**